config/envoyconfig/listeners.go

```diff
@@ -349,7 +365,7 @@ func (b *Builder) buildMainHTTPConnectionManagerFilter(
 		UseRemoteAddress:  &wrappers.BoolValue{Value: true},
 		SkipXffAppend:     options.SkipXffAppend,
 		XffNumTrustedHops: options.XffNumTrustedHops,
-		LocalReplyConfig:  b.buildLocalReplyConfig(options, requireStrictTransportSecurity),
+		LocalReplyConfig:  b.buildLocalReplyConfig(options, true),
```

Comment:
this appears to turn on HSTS headers for routes with self-signed certificates

Reply:
right, we would not set the HSTS part of `localReply` then, as it's applied on the virtual host level.

```json
{
  "localReplyConfig": {
    "mappers": [
      {
        "filter": {
          "responseFlagFilter": {}
        },
        "headersToAdd": [
          {
            "header": {
              "key": "X-Frame-Options",
              "value": "SAMEORIGIN"
            },
            "appendAction": "OVERWRITE_IF_EXISTS_OR_ADD"
          },
          {
            "header": {
              "key": "X-XSS-Protection",
              "value": "1; mode=block"
            },
            "appendAction": "OVERWRITE_IF_EXISTS_OR_ADD"
          }
        ]
      }
    ]
  }
}
```